runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -171,28 +176,52 @@ private void renderBonusOptions()
 	{
 		if (skillData.getBonuses() != null)
 		{
+			List<JCheckBox> uiCheckBoxList = new ArrayList<>();
+			lastBonus = 0.0f;
 			for (SkillDataBonus bonus : skillData.getBonuses())
 			{
 				JPanel uiOption = new JPanel(new BorderLayout());
 				JLabel uiLabel = new JLabel(bonus.getName());
-				JCheckBox uiCheckbox = new JCheckBox();
 
 				uiLabel.setForeground(Color.WHITE);
 				uiLabel.setFont(FontManager.getRunescapeSmallFont());
 
 				uiOption.setBorder(BorderFactory.createEmptyBorder(3, 7, 3, 0));
 				uiOption.setBackground(ColorScheme.DARKER_GRAY_COLOR);
 
-				// Adjust XP bonus depending on check-state of the boxes.
-				uiCheckbox.addActionListener(e -> adjustXPBonus(uiCheckbox.isSelected(), bonus.getValue()));
-				uiCheckbox.setBackground(ColorScheme.MEDIUM_GRAY_COLOR);
+				JCheckBox uiCheckBox = new JCheckBox();
+				uiCheckBoxList.add(uiCheckBox);
```

Comment:
Add the checkbox to the list after you're done messing with it, makes the code cleaner